src/Microsoft.AspNetCore.Sockets/Internal/AuthorizeHelper.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Sockets.Internal
+{
+    public static class AuthorizeHelper
+    {
+        public static async Task<bool> AuthorizeAsync(HttpContext context, AuthorizationPolicy policy)
```

Comment:
Never mind, `Task<bool>` is cached: https://referencesource.microsoft.com/#mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs,a431f9d7dd423518